### ISSUE TITLE
WebUI gets served incorrect config during node startup

### ISSUE THREAD

**Issue by ulope** [raiden-network/raiden#6072]
## Problem Definition

The WebUI loads it's configuration via an endpoint served by the Raiden node (`/ui/assets/config/config.<mode>.json`).

During node startup this information isn't available which leads to fallback code serving the html index page: https://github.com/raiden-network/raiden/blob/92ae3da956e5ccaead2681ac5c43f55efde6a1f7/raiden/api/rest.py#L394

Since the HTML content isn't valid configuration JSON the WebUI falls back to some built in configuration that tries to access a Web3 node on `localhost:8545`.

Since the fallback response has cache headers set to ~12h this leads to the WebUI being unusable even after the node has finished starting until the cache has expired.

**Comment by ulope** [comment]
#6073 partially fixes this by returning a proper json response during node startup to the WebUI on the config endpoint.

The behaviour should still be improved.
Several `Service Unavailable` errors are shown to the user while the node is starting and a manual reload is necessary afterwards.

**Comment by konradkonrad** [comment]
Still needs a manual reload in rc7.
--> `RETRY` works and tells user that `JSONRPCConnection` was established
--> but after retry the `Channels` tab still shows empty spinning circle. Manual page refresh fixes it.

**Comment by kelsos** [comment]
There is also this status endpoint now right? could the WebUI use that to monitor before starting to monitor the channels?

**Comment by manuelwedler** [comment]
The issue is that the WebUI needs the config for making the Web3 requests. At the moment there is this bug in the WebUI https://github.com/raiden-network/webui/issues/475, that makes it necessary to reload to retry the Web3 requests.

**Comment by ulope** [comment]
But does the WebUI need to make web3 requests before Raiden is done syncing?

There is a circular dependency in the API server which makes it difficult to generate the config file before the startup is finished.

**Comment by manuelwedler** [comment]
No, it does not. At the moment this is just causing that the user would have to reload the page. But I am fixing that on the WebUI now.

The benefit of having the config file would be that we would not show the RPC error on startup.

**Comment by kelsos** [comment]
@manuelwedler maybe it would make sense to avoid doing any web3 initialization before the status is available.

**Comment by manuelwedler** [comment]
I just implemented it that way that the Web3 initialization would get retried when the status is ready and it failed at initially.

I think the more common case is that the WebUI is loaded when the initial syncing of the client is done already. And it that case loading Web3 while the app initializes might be a bit faster than waiting for the app initialization and the response of the status endpoint.
